src/java/org/apache/cassandra/service/accord/FetchMinEpoch.java

```diff
@@ -145,21 +108,22 @@ public static Future<Long> fetch(SharedContext context, Map<InetAddressAndPort,
     }
 
     @VisibleForTesting
-    static Future<Long> fetch(SharedContext context, InetAddressAndPort to, Set<TokenRange> value)
+    static Future<Long> fetch(SharedContext context, InetAddressAndPort to)
     {
-        FetchMinEpoch req = new FetchMinEpoch(value);
-        return context.messaging().<FetchMinEpoch, FetchMinEpoch.Response>sendWithRetries(Backoff.NO_OP.INSTANCE,
-                                                                                          MessageDelivery.ImmediateRetryScheduler.instance,
-                                                                                          Verb.ACCORD_FETCH_MIN_EPOCH_REQ, req,
+        Backoff backoff = Backoff.fromConfig(context, DatabaseDescriptor.getAccord().minEpochSyncRetry);
+        return context.messaging().<FetchMinEpoch, FetchMinEpoch.Response>sendWithRetries(backoff,
+                                                                                          context.optionalTasks()::schedule,
+                                                                                          Verb.ACCORD_FETCH_MIN_EPOCH_REQ,
+                                                                                          FetchMinEpoch.instance,
                                                                                           Iterators.cycle(to),
-                                                                                          RetryPredicate.times(DatabaseDescriptor.getAccord().minEpochSyncRetry.maxAttempts.value),
+                                                                                          RetryPredicate.ALWAYS_RETRY,
```

Comment:
Does this mean that fetching the min epoch requires all nodes to be up to complete? Just looking at how this is accumulated by the caller of `fetch` which combines all the futures and can't complete until every future completes which means any down node would stop this from working?

Reply:
iirc this is faithful to the original implementation, which I believe might have been not entirely correct. We're collecting responses from all nodes, but we consider only successes here, which might mean we will not discover an early enough epoch. 

We planned to address this when implementing epoch GC, where we'll indicate which epoch is retired, and a better success criteria for this.

Reply:
I think that's not faithful to what was here, because it had a maximum number of attempts before so eventually it would complete? Now it retries forever so it will never stop?